training/tasks/prepare-training-set.js

```diff
@@ -67,10 +67,9 @@ class TypecheckStream extends Stream.Transform {
             return;
         }
 
-        const entities = Genie.Utils.makeDummyEntities(ex.preprocessed);
-        const program = ThingTalk.NNSyntax.fromNN(ex.target_code.split(' '), entities);
-
         try {
+            const entities = Genie.Utils.makeDummyEntities(ex.preprocessed);
+            const program = ThingTalk.NNSyntax.fromNN(ex.target_code.split(' '), entities);
```

Comment:
Got the following exception from makeDummyEntities 

Error: missing entity QUOTED_STRING_4
    at Object.makeDummyEntities (/opt/almond-cloud/node_modules/genie-toolkit/lib/utils.js:241:27)
    at TypecheckStream.process (/opt/almond-cloud/training/tasks/prepare-training-set.js:70:38)
    at TypecheckStream._transform (/opt/almond-cloud/training/tasks/prepare-training-set.js:83:14)
    at TypecheckStream.Transform._read (_stream_transform.js:190:10)
    at TypecheckStream.Transform._write (_stream_transform.js:178:12)
    at doWrite (_stream_writable.js:415:12)
    at clearBuffer (_stream_writable.js:545:7)
    at onwrite (_stream_writable.js:470:7)
    at TypecheckStream.afterTransform (_stream_transform.js:94:3)
    at process.then (/opt/almond-cloud/training/tasks/prepare-training-set.js:83:37)

Reply:
It's a genie bug, triggered by some bad sentences in the database. I deleted them manually.